File: src/frontend/libs/data-access/src/lib/models/reservation/reservation-details.dto.ts
/**
 * Vehicle and reservation details
 */
import type { ISODateString } from '../common';
import type { VehicleId, CategoryCode } from '../vehicle';
import type { LocationCode } from '../location';

export type ReservationDetails = {
  readonly vehicleId: VehicleId;
  readonly categoryCode: CategoryCode;
  readonly pickupDate: ISODateString;
  readonly returnDate: ISODateString;
  readonly pickupLocationCode: LocationCode;
  readonly dropoffLocationCode: LocationCode;
};

File: src/backend/Services/Reservations/OrangeCarRental.Reservations.Api/Shared/ReservationDetailsDto.cs
namespace SmartSolutionsLab.OrangeCarRental.Reservations.Api.Shared;

/// <summary>
///     Vehicle and reservation details.
/// </summary>
public sealed record ReservationDetailsDto(
    Guid VehicleId,
    string CategoryCode,
    DateOnly PickupDate,
    DateOnly ReturnDate,
    string PickupLocationCode,
    string DropoffLocationCode);
